version: '3.8'

services:
  db:
    image: postgres:latest
    environment:
      - POSTGRES_DB=MyDatabase
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
    ports:
      - "5432:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U postgres" ]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - network
  
  jaeger:
    image: jaegertracing/all-in-one:latest
    ports:
      - "16686:16686"  # UI
      - "4317:4317"    # OTLP gRPC
      - "4318:4318"    # OTLP HTTP
      - "14250:14250"  # Model
      - "14268:14268"  # Jaeger HTTP
      - "6831:6831/udp"  # Jaeger thrift
    environment:
      - COLLECTOR_OTLP_ENABLED=true
    networks:
      - network      
  
volumes:
  postgres_data:

networks:
  network:
    driver: bridge